bin/magpie-hive-daemon.sh

```diff
@@ -0,0 +1,208 @@
+#  Copyright (C) 2013-2015 Lawrence Livermore National Security, LLC.
+#  Produced at Lawrence Livermore National Laboratory (cf, DISCLAIMER).
+#  Written by Albert Chu <chu11@llnl.gov>
+#  LLNL-CODE-644248
+#  
+#  This file is part of Magpie, scripts for running Hadoop on
+#  traditional HPC systems.  For details, see https://github.com/llnl/magpie.
+#  
+#  Magpie is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#  
+#  Magpie is distributed in the hope that it will be useful, but
+#  WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  General Public License for more details.
+#  
+#  You should have received a copy of the GNU General Public License
+#  along with Magpie.  If not, see <http://www.gnu.org/licenses/>.
+#############################################################################
+
+# This script launches Hive for the user
+
+# First argument is path to Hive Configuration directory
+
+# Second argument is path to Hive binary
+
+# Third argument is start or stop
+
+if [ "$1X" == "X" ]
+then
+    echo "User must specify Hive configuration path"
+    exit 1
+fi
+
+if [ ! -d $1 ]
+then
+    echo "Hive configuration path is not a directory"
+fi
+
+if [ "$2X" == "X" ]
+then
+    echo "User must specify path to Hive as second argument"
+    exit 1
+fi
+
+if [ ! -d $2 ]
+then
+    echo "Hive path is not a directory"
+fi
+
+if [ "$3X" == "X" ]
+then
+    echo "User must specify start or stop as third argument"
+    exit 1
+fi
+
+if [ "$3" != "start" ] && [ "$3" != "stop" ]
+then
+    echo "User must specify start or stop as third argument"
+    exit 1
+fi
+
+# source the hive-env.sh file
+source $1/hive-env.sh
+myhostname=`hostname`
+
+if [ "$3" == "start" ]
+then
+    if [ -f "${HIVE_PID_DIR}/hive-metastore.pid" ]
+    then
+        echo "Hive metastore pid file found, it should already be running"
+    else
+        if [ "${HIVE_METASTORE_DIR}X" != "X" ]
+        then
+            hivemetastoredir=${HIVE_METASTORE_DIR}
+        else
+            hivemetastoredir="${hivelocalscratchdirpath}/hive/database/postgres/data"
+        fi
+       
+        $POSTGRES_HOME/bin/pg_isready -h localhost -p $PGPORT &> /dev/null &
+        postgrescheck=$?
+
+        if [[ $postgrescheck != 0 ]] 
+        then
+          echo "pg_isready reports: ${postgrescheck} -- Postgres not ready on ${PGPORT}.  Exiting Hive."
+          exit 1
+        fi
+        # check for the metastore warehouse directory, if not present create it.
+        if [ ! -d $HIVE_METASTORE_DIR ]
+        then
+            echo "Metastore warehouse directory ( ${HIVE_METASTORE_DIR} ) not found, creating directory now."
+            mkdir -p $HIVE_METASTORE_DIR
```

Comment:
should check for error on mkdir and exit if needed

Reply:
Added